saleor/graphql/product/tests/test_attributes.py

```diff
@@ -802,22 +802,29 @@ def test_assignment_attribute_update_assigned_unsupported_type_should_raise_an_e
         assert error in expected_errors
 
 
+@pytest.mark.parametrize("api_client", ["app_api_client", "staff_api_client"])
 def test_assignment_attribute_update_assigned_should_modify_variant_selection(
-    staff_api_client,
+    api_client,
     permission_manage_product_types_and_attributes,
     size_attribute,
     color_attribute_without_values,
+    request,
 ):
     """The productAttributeAssignmentUpdate mutation should modify
     variant selection when validation successful."""
-
+    chosen_api_client = request.getfixturevalue(api_client)
     product_type = ProductType.objects.create(name="Type", kind=ProductTypeKind.NORMAL)
     attribute_1 = color_attribute_without_values
     attribute_2 = size_attribute
 
-    staff_api_client.user.user_permissions.add(
-        permission_manage_product_types_and_attributes
-    )
+    if api_client == "staff_api_client":
+        chosen_api_client.user.user_permissions.add(
+            permission_manage_product_types_and_attributes
+        )
+    else:
+        chosen_api_client.app.permissions.add(
+            permission_manage_product_types_and_attributes
+        )
```

Comment:
In such cases, we prefer having two separate tests than having if-s in tests. To avoid code duplication you could extract a function (if this makes sense here).

Reply:
I'll just make two separate tests, though.

Reply:
Done, I have decided, that it does not have make sense to extract a function at this point (if there will be 3+ tests,  I'd consider it).